src/phase0/state_transition/block_processing.rs

```diff
@@ -160,16 +165,69 @@ fn process_block_header<
         MAX_VALIDATORS_PER_COMMITTEE,
         PENDING_ATTESTATIONS_BOUND,
     >,
-    _block: &BeaconBlock<
+    block: &mut BeaconBlock<
         MAX_PROPOSER_SLASHINGS,
         MAX_VALIDATORS_PER_COMMITTEE,
         MAX_ATTESTER_SLASHINGS,
         MAX_ATTESTATIONS,
         MAX_DEPOSITS,
         MAX_VOLUNTARY_EXITS,
     >,
-    _context: &Context,
+    context: &Context,
 ) -> Result<(), Error> {
+    if block.slot != state.slot {
+        return Err(Error::InvalidOperation(InvalidOperation::Header(
+            InvalidBeaconBlockHeader::StateSlotMismatch {
+                state_slot: state.slot,
+                block_slot: block.slot,
+            },
+        )));
+    }
+
+    if block.slot <= state.latest_block_header.slot {
+        return Err(Error::InvalidOperation(InvalidOperation::Header(
+            InvalidBeaconBlockHeader::OlderThanLatestBlockHeader {
+                block_slot: block.slot,
+                latest_block_header_slot: state.latest_block_header.slot,
+            },
+        )));
+    }
+
+    let proposer_index = get_beacon_proposer_index(state, context)?;
+    if block.proposer_index != proposer_index {
+        return Err(Error::InvalidOperation(InvalidOperation::Header(
+            InvalidBeaconBlockHeader::ProposerIndexMismatch {
+                block_proposer_index: block.proposer_index,
+                state_proposer_index: proposer_index,
```

Comment:
i think in this context `state` is redundant so in favor of reducing verbosity:

```suggestion
                proposer_index,
```

Reply:
Done:  24b9f6e